scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/language/json/JsonTemplateParser.scala

```diff
@@ -0,0 +1,181 @@
+        TypedExpression(
+          CompiledJsonTemplateExpression(languageId, original, templateTypeExpression.expression, expectedType),
+          JsonTemplateExpressionTypingInfo(jsonTypeExpression.typingInfo, expectedType),
+        )
+      }
+
+  override def parseWithoutContextValidation(
+      original: String,
+      expectedType: typing.TypingResult
+  ): ValidatedNel[ExpressionParseError, CompiledExpression] = spelTemplateParser
+    .parseWithoutContextValidation(original, stringTypingResult)
+    .andThen(parsedSpelTemplateExpression =>
+      extractJsonStringFromCompiledExpression(parsedSpelTemplateExpression)
+        .andThen(jsonString => JsonParser.parseWithoutContextValidation(jsonString, expectedType))
+        .map(_ => parsedSpelTemplateExpression)
+    )
+    .map(templateTypeExpression =>
+      CompiledJsonTemplateExpression(languageId, original, templateTypeExpression, expectedType)
+    )
+
+  private def extractJsonStringFromParsedExpression(
+      typedExpression: TypedExpression,
+      ctx: ValidationContext,
+  ): ValidatedNel[ExpressionParseError, String] =
+    typedExpression.expression match {
+      case expression: SpelExpression =>
+        joinLiteralsAndReplaceExpressionWithDefaults(
+          spelExpressionToDefaultJsonValue(ctx),
+          expression.parsedSpringExpression,
+        )
+      case _ => invalidNel(ExpressionCompilationError("Invalid compiled expression type"))
+    }
+
+  private def joinLiteralsAndReplaceExpressionWithDefaults(
+      spelExpressionToDefaultJsonValue: (String, typing.TypingResult) => ValidatedNel[ExpressionParseError, String],
+      expression: SpringExpression,
+  ): ValidatedNel[ExpressionParseError, String] = expression match {
+    case expression: CompositeStringExpression =>
+      expression.getExpressions.toList
+        .map(e => joinLiteralsAndReplaceExpressionWithDefaults(spelExpressionToDefaultJsonValue, e))
+        .sequence
+        .map(_.mkString)
+    case expression: LiteralExpression => validNel(expression.getValue)
+    case expression: standard.SpelExpression =>
+      spelExpressionToDefaultJsonValue(expression.getExpressionString, Typed[SpelExpressionRepr])
+    case _ => invalidNel(ExpressionCompilationError("Unknown expression type"))
+  }
+
+  private def spelExpressionToDefaultJsonValue(ctx: ValidationContext)(
+      original: String,
+      expectedType: typing.TypingResult
+  ): ValidatedNel[ExpressionParseError, String] = spelParser
+    .parse(original, ctx, expectedType)
+    .map(_.typingInfo.typingResult)
+    .map(typingResultToDefaultJsonValue)
+
+  private def typingResultToDefaultJsonValue(typingResult: TypingResult): String = typingResult match {
+    case TypedClass(k, _) =>
+      k match {
+        case className if TypeValueDeterminer.isIntegerNumber(className)       => "0"
+        case className if TypeValueDeterminer.isFloatingPointNumber(className) => "0.5"
```

Comment:
Magic number. Let's extract a well-named constant for that e.g. `placeholderForFloatingPoint` with the comment that the value is not important, the only requirement that it should fulfil is that it can't be rounded.